ingestion/src/metadata/ingestion/ometa/mixins/role_policy_mixin.py

```diff
@@ -0,0 +1,440 @@
+#  Copyright 2023 Schlameel
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+Mixin class containing Role and Policy specific methods
+
+To be used by OpenMetadata class
+"""
+import json
+import traceback
+from typing import Dict, Generic, List, Optional, Type, TypeVar, Union
+
+from pydantic import BaseModel
+
+from metadata.generated.schema.entity.policies.accessControl.rule import Rule
+from metadata.generated.schema.entity.policies.policy import Policy
+from metadata.generated.schema.entity.teams.role import Role
+from metadata.generated.schema.type import basic
+from metadata.ingestion.ometa.client import REST
+from metadata.ingestion.ometa.utils import model_str
+from metadata.utils.logger import ometa_logger
+
+logger = ometa_logger()
+
+T = TypeVar("T", bound=BaseModel)
+
+OPERATION = "op"
+PATH = "path"
+VALUE = "value"
+VALUE_ID = "id"
+VALUE_TYPE = "type"
+VALUE_POLICY = "policy"
+VALUE_CONDITION: str = "condition"
+VALUE_DESCRIPTION: str = "description"
+VALUE_EFFECT: str = "effect"
+VALUE_FQN: str = "fullyQualifiedName"
+VALUE_NAME: str = "name"
+VALUE_OPERATIONS: str = "operations"
+VALUE_RESOURCES: str = "resources"
+
+# Operations
+ADD = "add"
+REPLACE = "replace"
+REMOVE = "remove"
+
+# Paths
+ROLES_PATH: str = "/roles/{role_id}"
```

Comment:
Thinking if it might be easier to read and use it if we just wrap all these constants in an `Enum`?

Reply:
We can name the Enums as

```python
class PatchField(str, Enum):
    OPERATION = "op"
    PATH = "path"
    VALUE = "value"
```